Sources/Caching/Caching.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Caching.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// An inteface representing a simple cache
+@_spi(Internal) public protocol Caching {
+
+    /// A URL for a cache directory if one is present
+    var cacheDirectory: URL? { get }
```

Comment:
I wonder why we need the directory to be part of the protocol. This feels like implementation details to me.